### ISSUE TITLE
wings service does not start

### ISSUE THREAD

**Issue by Xedon420** [pterodactyl/panel#2448]
* Panel or Daemon: Daemon
* Version of Panel/Daemon: 1.0.0-rc.7
* Server's OS: Debian 9
* Your Computer's OS & Browser: elementary os, firefox

Panel: `php -v` PHP 7.4.10 (cli) (built: Sep  9 2020 06:55:12) ( NTS )
Daemon: 4.9.0-13-amd64 #1 SMP Debian 4.9.228-1 (2020-07-05) x86_64 GNU/Linux

**To Reproduce**
1. upgrade from 0.6 Daemon and 0.7 Panel to 1.0
2. start wings service

- I have already recreated the config.yml
- Here are the wing logs: https://bin.ptdl.co/7yfjj/
- `allowed_mounts: [] and 
allowed_origins: []` is correct in my config.yml

the error:
`FATAL: [Oct  4 11:57:56.398] failed to load server configurations error=json: cannot unmarshal number into Go struct field .process_configuration.startup.done of type []*api.OutputLineMatcher`

**Comment by Xedon420** [comment]
I have now reset the server via backup to the older daemon version and panel version.

I found out that a newly created node works fine with the daemon.
It seems that there is something wrong in the migrate process from panel or daemon to 1.0, because it only affects the node that existed before the migrate.

I will now wait until the stable 1.0 comes out

**Comment by DaneEveritt** [comment]
Do you have an egg whose "Done" value is simply a number or something?

**Comment by DaneEveritt** [comment]
Closing this until enough information to properly reproduce this issue is provided.

**Comment by lukas-fichtner** [comment]
The problem still exists.
How can I view the eggs in migrated state?
I'm really stuck here, another downtime of game servers with the same problem....

**Comment by DaneEveritt** [comment]
Look through all of the servers you have configured and provide links to whatever _non-default_ eggs are being used.

**Comment by lukas-fichtner** [comment]
Fortunately, I was able to solve the problem now.
I simply restored a full backup of my game host and BEFORE the migration deleted all the eggs that I don't currently have in use.
I also removed the brackets in the startup config of the spigot egg.

(After that I removed php completely and installed php 7.4, as before)
After that I followed the upgrade guide and finally the wings service was ready to start.
Afterwards all game servers are crashed.
I have now changed the network_name & network_mode in config.yml to "host" (like in the old daemon version).
I ask for correction if this is wrong. Now at least all game servers start again.
Now it looks like everything is running. Tomorrow I will give a feedback if there are any problems again.

But I can generally recommend (if not already standard) to make a fullstack backup before the upgrade. Something might still go wrong during the upgrade!

**Comment by DaneEveritt** [comment]
The only thing I needed to know was what egg was returning "bad" data to wings. But good to know. Reopening this since I'm pretty sure this is a trivial code fix based on the stack.

**Comment by lukas-fichtner** [comment]
Yes, I can confirm that it must have been due to an incorrect egg setting / parameter.
In any case, it must have been an egg that I am not actively using because I did not delete or change it. Which one I unfortunately do not know exactly. It must have been one of the default eggs, maybe it was broken during a previous upgrade, but it was not noticed because it was not in use. There are many possibilities here.
Maybe it would be possible to validate the egg configs before migration.
This would possibly avoid such an error in the future.
